pyccel/parser/semantic.py

```diff
@@ -1132,6 +1136,73 @@ def _handle_function(self, expr, func, args, *, is_method = False, use_build_fun
                             func, func.is_elemental)
 
             return new_expr
+    def _handle_kernel(self, expr, func, args, **settings):
+        """
+        Create the node representing the kernel function call.
+
+        Create a FunctionCall or an instance of a PyccelInternalFunction
+        from the function information and arguments.
+
+        Parameters
+        ----------
+        expr : IndexedFunctionCall
+               Node has all the information about the function call
+
+        func : FunctionDef instance, Interface instance or PyccelInternalFunction type
+               The function being called.
+
+        args : tuple
+               The arguments passed to the function.
+
+        **settings : dict
+            The settings passed to _visit functions.
+
+        Returns
+        -------
+        Pyccel.ast.cuda.KernelCall
+            The semantic representation of the kernel call.
+        """
+        if(len(expr.launch_config) < 2):
+            errors.report(INVALID_KERNEL_LAUNCH_CONFIG_LOW,
+                    symbol=expr,
+                    severity='fatal')
+        if(len(expr.launch_config) > 2):
+            errors.report(INVALID_KERNEL_LAUNCH_CONFIG_HIGH,
+                    symbol=expr,
+                    severity='fatal')
```

Comment:
These error messages refer to types with missing methods but you don't seem to be checking the type here. How do you know that integer indexes are used?

Reply:
So basically, the error arises from Numba when you pass the wrong number of indexes to your kernel call. For example, if you pass just one number instead of two, it doesn't make a tuple for them when trying to access the method length in the field. Should I change it to something simpler

Reply:
I'm not sure hwat the relevance of Numba is. The error here is raised by Pyccel. It should tell the user how to modify their code to avoid the error. It shouldn't blindly reproduce the error that you hope you would see in Python here.

The errors that you are raising are:
> "'int' object has no attribute 'handle'"

and
> "object of type 'int' has no len()"

If I did:
```python
f[(1,2)]()
```
Then I would see the error "object of type 'int' has no len()". This would make no sense to me here. This is because you are assuming that I have written:
```python
f[1]()
```
but this is not what you are catching.

I think you should rewrite the error message to describe what you are testing (i.e the number of parameters in the launch configuration)

Reply:
can u check now  adad61a

Reply:
Much clearer :+1: